gadget/install/install.go

```diff
@@ -131,7 +131,16 @@ func maybeEncryptPartition(laidOut *gadget.LaidOutStructure, encryptionType secb
 			timings.Run(perfTimings, fmt.Sprintf("new-encrypted-device[%s]", laidOut.Role()),
 				fmt.Sprintf("Create encryption device for %s", laidOut.Role()),
 				func(timings.Measurer) {
-					dataPart, err = newEncryptedDeviceLUKS(&laidOut.OnDiskStructure, encryptionKey, laidOut.PartitionFSLabel, laidOut.Name())
+					dataPart, err = newEncryptedDeviceLUKS(&laidOut.OnDiskStructure, encryptionType, encryptionKey, laidOut.PartitionFSLabel, laidOut.Name())
+				})
+			if err != nil {
+				return nil, nil, err
+			}
```

Comment:
there are more places that look for EncryptionTypeLUKS here that I think also need changes. Related we also will need to support this in the new install APIs instead of just auto-install, we hard code LUKS in a couple of places there. I would recommend to grep around the codebase for EncryptionTypeLUKS to see what might need changes and either fix it here or leave TODOs

Reply:
I updated them and added a bunch of TODOs, given that this is needed soon I'm inclined to keep this PR small and do a followup that clear out all the device-setup-hook work that is no longer needed with this new ICE approach.